### PR TITLE
(#579) External Link Icon in Buttons

### LINKED ISSUE #579: Enabler: Update Buttons to handle external (non federal website) links
This will comprise only updating `usa-button` when it is applied to an anchor element. Use the existing USWDS Launch icon for the external link. This is not a button variant. 

## Tasks
- [ ] Icon should automatically be added if .gov is not found in the url. 
  - dev note: look into css regex
- [ ] Icon should be able to be forced to show or forced to be suppressed
- [ ] Storybook: create stories with external links. All variants?
- [ ] Example JS site: only updated if JS used
- [ ] Documentation site: update guidance. Needs guidance?

Per discussion with ODDC Feds on 11/8:
- received approval to move forward with CSS approach to trigger the display of exist disclaimer
- exit disclaimer should display on not US Federal sites (sites not ending in .gov)
- exit disclaimer should not display on the following domains (create a rule to not show exit disclaimer): https://public.govdelivery.com/*, https://www.gov.uk/*

### PR DESCRIPTION
**Pull request by karlikpj**
Closes #579

### CODE REVIEW

**Comment by olitharp-nci** [comment]
Remove parenthesis in description so github links to ticket `Closes #579`

File: packages/ncids-css/scss/components/usa-button/_all.scss [hunk 1]
```diff
@@ -7,3 +7,4 @@
 
 // nci
 @import 'internal/nci-small';
+@import 'internal/external';
```
Comment (olitharp-nci) [inline packages/ncids-css/scss/components/usa-button/_all.scss:10]: add new line at end of file

**Comment by blilianyu** [comment]
@karlikpj  Can we increase margin-left from 1px to 10px? The icon is too tight
Screen Shot 2022-10-17 at 1 46 04 PM

File: testing/ncids-css-testing/stories/components/usa-button/usa-button.stories.jsx [hunk 2]
```diff
@@ -13,3 +13,4 @@ export * from "./outline-inverse.stories";
 export * from "./secondary.stories";
 export * from "./unstyled.stories";
 export * from "./nci-small.stories";
+export * from "./external-link.stories";
```
Comment (olitharp-nci) [inline testing/ncids-css-testing/stories/components/usa-button/usa-button.stories.jsx:16]: new line at end of file

**Review by olitharp-nci — CHANGES_REQUESTED** [review]
We need guidance for this variant so the documentation site can be updated

**Comment by karlikpj** [comment]
@blilianyu can you please update comment to use one of the size tokens from https://designsystem.digital.gov/design-tokens/spacing-units/
(1 or 1.5 i feel would be best)

**Review by monika-jaeger — APPROVED** [review]
Looks good after the increased space!

**Review by sarinapadilla — APPROVED** [review]
This passes product review!

**Review by olitharp-nci — APPROVED** [review]

**Review by dlescarbeau — APPROVED** [review]

**Review by bryanpizzillo — CHANGES_REQUESTED** [review]
Please add code and documentation so that:
1. We can force showing the external link if it matches a ".gov" address. (E.g. it matches https://www.govgrantfinder.com)
2. We can force NOT showing the external link if the link is a URL we do not consider to be external such as someproject.ideascale.com.

**Review by olitharp-nci — APPROVED** [review]

**Review by arcepaul — APPROVED** [review]

**Pull request merged**